expression/builtin_json.go

```diff
@@ -702,8 +703,102 @@ type jsonArrayAppendFunctionClass struct {
 	baseFunctionClass
 }
 
+type builtinJSONArrayAppendSig struct {
+	baseBuiltinFunc
+}
+
 func (c *jsonArrayAppendFunctionClass) getFunction(ctx sessionctx.Context, args []Expression) (builtinFunc, error) {
-	return nil, errFunctionNotExists.GenWithStackByArgs("FUNCTION", "JSON_ARRAY_APPEND")
+	if err := c.verifyArgs(args); err != nil {
+		return nil, err
+	}
+	if len(args)&1 != 1 {
+		return nil, ErrIncorrectParameterCount.GenWithStackByArgs(c.funcName)
+	}
+	argTps := make([]types.EvalType, 0, len(args))
+	argTps = append(argTps, types.ETJson)
+	for i := 1; i < len(args)-1; i += 2 {
+		argTps = append(argTps, types.ETString, types.ETJson)
+	}
+	bf := newBaseBuiltinFuncWithTp(ctx, args, types.ETJson, argTps...)
+	for i := 2; i < len(args); i += 2 {
+		DisableParseJSONFlag4Expr(args[i])
+	}
+	sig := &builtinJSONArrayAppendSig{bf}
+	sig.setPbCode(tipb.ScalarFuncSig_JsonArrayAppendSig)
+	return sig, nil
+}
+
+func (b *builtinJSONArrayAppendSig) Clone() builtinFunc {
+	newSig := &builtinJSONArrayAppendSig{}
+	newSig.cloneFrom(&b.baseBuiltinFunc)
+	return newSig
+}
+
+func (b *builtinJSONArrayAppendSig) evalJSON(row chunk.Row) (res json.BinaryJSON, isNull bool, err error) {
+	res, isNull, err = b.args[0].EvalJSON(b.ctx, row)
+	if err != nil {
+		b.ctx.GetSessionVars().StmtCtx.AppendWarning(json.ErrInvalidJSONPath.FastGen("Syntax error in JSON text in argument 1 to function 'json_set'"))
```

Comment:
Does MySQL return error to the client instead of warning when it's in strict sql mode?

Reply:
Please use the pre-defined error message pattern in "github.com/pingcap/parser/mysql"

Reply:
> Does MySQL return error to the client instead of warning when it's in strict sql mode?

```
MariaDB [(none)]> SELECT @@GLOBAL.sql_mode;
+-------------------------------------------------------------------------------------------------------------+
| @@GLOBAL.sql_mode                                                                                           |
+-------------------------------------------------------------------------------------------------------------+
| STRICT_TRANS_TABLES,STRICT_ALL_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_AUTO_CREATE_USER,NO_ENGINE_SUBSTITUTION |
+-------------------------------------------------------------------------------------------------------------+
1 row in set (0.000 sec)

MariaDB [(none)]> 
MariaDB [(none)]> SELECT JSON_ARRAY_APPEND('{}', "asd", "w");
+-------------------------------------+
| JSON_ARRAY_APPEND('{}', "asd", "w") |
+-------------------------------------+
| NULL                                |
+-------------------------------------+
1 row in set, 1 warning (0.000 sec)

MariaDB [(none)]> show warnings ;
+---------+------+---------------------------------------------------------------------------------------+
| Level   | Code | Message                                                                               |
+---------+------+---------------------------------------------------------------------------------------+
| Warning | 4042 | Syntax error in JSON path in argument 2 to function 'json_array_append' at position 1 |
+---------+------+---------------------------------------------------------------------------------------+
1 row in set (0.000 sec)
```

Seems true.